### PR TITLE
[MM-23482] Make team list iterate until it gets all server teams

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
The `team list` command was querying for one page of teams with a `per-page` param of 10000 teams. The API has a limit of 200 for the `per-page` params, making the command only show the first 200.

This PR makes the command iterate over several teams pages, printing all the server teams.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23482

### CODE REVIEW

File: commands/team.go [hunk 1]
```diff
@@ -176,17 +177,26 @@ func archiveTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error
...
-			printer.PrintT("{{.Name}}", team)
+		for _, team := range teams {
+			if team.DeleteAt > 0 {
+				printer.PrintT("{{.Name}} (archived)", team)
+			} else {
+				printer.PrintT("{{.Name}}", team)
+			}
 		}
+
+		if len(teams) < web.LIMIT_MAXIMUM {
```
Comment (wiggin77) [inline commands/team.go]: Looks good. I double checked that teams endpoint doesn't error on empty page, otherwise if number of teams total was a multiple of `web.LIMIT_MAXIMUM` it would return error.  Instead, in that case, it will do one last request, get an empty list then exit clean.

Comment (wiggin77) [reply commands/team.go]: Maybe add a unit test for team count a multiple of `web.LIMIT_MAXIMUM` just in case the endpoint (e.g. v6) starts erroring for page numbers beyond the last one?  Not sure if necessary.

File: commands/team.go [hunk 2]
```diff
@@ -176,17 +177,26 @@ func archiveTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error
 }
 
 func listTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
-	teams, response := c.GetAllTeams("", 0, 10000)
-	if response.Error != nil {
-		return response.Error
-	}
+	page := 0
+	for {
+		teams, response := c.GetAllTeams("", page, web.LIMIT_MAXIMUM)
```
Comment (jespino) [inline commands/team.go]: ok, importing this `web.LIMIT_MAXIMUM` constant is triggering here the inclusion in the vendor directory a zillion of new files. I think can be more interesting to have a copy of the constant and avoid all that vendors directory explosion.

File: commands/team.go (hunk 1 above)
Comment (mgdelacroix) [reply commands/team.go]: Hum... it's an interesting corner case, but I think that if the API would start erroring on empty pages, we would treat those errors as bugs, so the test case wouldn't be needed. I can add it if you feel strongly about it tho
